File: formula-one/webapp/src/app/services/formula-one.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { FormulaOneItem } from '../models/formula-one-item';
import { DeleteResponse } from '../models/delete-response';

@Injectable({
  providedIn: 'root',
})
export class FormulaOneService {
  private readonly API: string = 'http://localhost:8080/api/f1';

  constructor(private readonly http: HttpClient) {}

  getTeams(): Observable<FormulaOneItem[]> {
    return this.http.get<FormulaOneItem[]>(this.API);
  }

  addTeam(item: FormulaOneItem): Observable<FormulaOneItem> {
    return this.http.post<FormulaOneItem>(`${this.API}/team`, item);
  }

  updateTeam(item: FormulaOneItem): Observable<FormulaOneItem> {
    return this.http.put<FormulaOneItem>(`${this.API}/team`, item);
  }

  deleteTeam(id: string): Observable<DeleteResponse> {
    return this.http.delete<DeleteResponse>(`${this.API}/team/${id}`);
  }
}

File: formula-one/src/main/java/org/demo/service/FormulaOneService.java
package org.demo.service;

import org.apache.coyote.BadRequestException;
import org.demo.model.DeleteResponse;
import org.demo.model.FormulaOneItem;

import java.util.List;

public interface FormulaOneService {
    List<FormulaOneItem> getTeams();

    FormulaOneItem addTeam(FormulaOneItem formulaOneItem) ;

    FormulaOneItem updateTeam(FormulaOneItem formulaOneItem);

    DeleteResponse deleteTeam(String id);
}
